driftbase/api/richpresence.py

```diff
@@ -0,0 +1,67 @@
+"""
+Rich Presence is player metadata which is relevant to others, such as a players online status. Rich presence
+can be queried from this extension, or you can listen to the message queue to get live updates without polling.
+"""
+
+from flask.views import MethodView
+from driftbase.flask.flaskproxy import g
+from driftbase.models.db import CorePlayer
+from marshmallow import Schema, fields
+from drift.blueprint import Blueprint
+import http.client as http_client
+from driftbase.richpresence import RichPresenceSchema, get_richpresence
+from drift.core.extensions.urlregistry import Endpoints
+from flask import url_for
+from webargs.flaskparser import abort
+
+
+bp = Blueprint("richpresence", __name__, url_prefix="/rich-presence/")
+endpoints = Endpoints()
+
+def drift_init_extension(app, **kwargs):
+    app.register_blueprint(bp)
+    endpoints.init_app(app)
+
+class RichPresenceRequestSchema(Schema):
+    name = fields.List(fields.String())
+
+class RichPresenceListArgs(Schema):
+    class Meta:
+        strict = True
+
+    player_id = fields.List(
+        fields.Integer(), metadata=dict(description="Player ID's to filter for"
+    ))
+
+@bp.route('/<int:player_id>', endpoint='entry')
+class RichPresenceAPI(MethodView):
+    @bp.response(http_client.OK, RichPresenceSchema)
+    def get(self, player_id : int):
+        """
+        Single Player
+
+        Retrieve rich-presence information for a specific player
+        """
+        
+        player = g.db.query(CorePlayer).get(player_id)
```

Comment:
I'd suggest leaving the player check to the service code, throw an exception and turn it into a 404 here, instead of having partial logic in the API handler. drift-base does not follow the rules very well, but API handlers should really only concern themselves with validating the input follows approved syntax, and formatting the responses, they should have no business logic in them. Again, not a pattern drift-base follows in general, but since this is new code, why not.

Reply:
I understand what you mean. What's the best way of percolating an error from the service to the API handler?  Let's say I want to return `NOT_FOUND` in some cases, instead of always `INTERNAL_SERVER_ERROR`. 

I can either return a tuple of err/value, or throw+handle exceptions? Any preferences?